### DISCUSSION TITLE
Windows10 Installation: Failed to run `west update`

### DISCUSSION THREAD

**Discussion by r0n9Y** [zephyrproject-rtos/zephyr#48837] [General]
**Failed to run `west update` when I try to install Zephyr SDK first time**
with the error `fatal: a54986aa98db4082ac56b582843bb5b5435208a6^commit: not a valid SHA1`
Here is details of the error:
```
=== updating zscilib (modules/lib/zscilib):
west.manifest: zscilib: checking if cloned
west.manifest: running 'git rev-parse --show-cdup' in C:\Users\sanju\zephyr_project\modules\lib\zscilib
west.manifest: "git rev-parse --show-cdup" exit code: 0 stdout: b'\n' stderr: b''
west.manifest: running 'git cat-file -t a54986aa98db4082ac56b582843bb5b5435208a6' in C:\Users\sanju\zephyr_project\modules\lib\zscilib
west.manifest: "git cat-file -t a54986aa98db4082ac56b582843bb5b5435208a6" exit code: 0 stdout: b'commit\n' stderr: b''
west.manifest: running 'git rev-parse --verify --symbolic-full-name a54986aa98db4082ac56b582843bb5b5435208a6' in C:\Users\sanju\zephyr_project\modules\lib\zscilib
west.manifest: "git rev-parse --verify --symbolic-full-name a54986aa98db4082ac56b582843bb5b5435208a6" exit code: 0 stdout: b'' stderr: b''
skipping unnecessary fetch
west.manifest: running 'git update-ref -m 'west update: moving to a54986aa98db4082ac56b582843bb5b5435208a6^{commit}' refs/heads/manifest-rev 'a54986aa98db4082ac56b582843bb5b5435208a6^{commit}'' in C:\Users\sanju\zephyr_project\modules\lib\zscilib
fatal: a54986aa98db4082ac56b582843bb5b5435208a6^commit: not a valid SHA1
west.manifest: "git update-ref -m 'west update: moving to a54986aa98db4082ac56b582843bb5b5435208a6^{commit}' refs/heads/manifest-rev 'a54986aa98db4082ac56b582843bb5b5435208a6^{commit}'" exit code: 128 stdout: None stderr: None
ERROR: update failed for multiple projects; see above
```
 - Zephyr version: 3.1.99
 - Following the installation guide for window from https://docs.zephyrproject.org/latest/develop/getting_started/index.html
 - Unable to get the Zephyr source code
 - env: Windows 10 version: 21H2 19044.1826
 - git version 2.36.1

**Question**
Is it possible I can find pre-built binary somewhere? It is unusual that the ppl have to build the SDK from scratch. And I am trying to avoid to use PlatformIO with Zephyr.

**Comment by carlescufi** [comment]
@r0n9Y which west version are you using? `west --version`

**Comment by carlescufi** [comment]
> Is it possible I can find pre-built binary somewhere? It is unusual that the ppl have to build the SDK from scratch. And I am trying to avoid to use PlatformIO with Zephyr.

You do not need to build the SDK from scratch, this is Zephyr and its source code and not the SDK.

**Comment by r0n9Y** [comment]
West version: v0.13.1

**Comment by mbolivar-nordic** [comment]
@r0n9Y can you say which shell you are running on? We only support cmd.exe on windows, so if you're using the git bash shell, please retry on cmd.exe.

**Comment by TomzBench** [comment]
@mbolivar-nordic 

Per our discord chat, here are the logs:

west -v update logs for atx-zdk
https://gist.github.com/TomzBench/73722f7b2cb3d0feeb58b0b56c5ecf7b

west -v update logs for zephyr
https://gist.github.com/TomzBench/2de8fd39d140cddd9cddff04f367bab3

**Reply by mbolivar-nordic** [reply]
Thank you @tomzbench, and you as well @r0n9Y, those are useful logs. I was able to take a look at them and I got an idea; could either of you please try this PR and see if your issue goes away?

https://github.com/zephyrproject-rtos/west/pull/599

You can install with:

```
pip3 install git+https://github.com/mbolivar-nordic/west@project-git-on-windows-hotfix 
```

Then test again.

To go back to another version after you're done testing, e.g. v0.14.0, you can do this:

```
pip3 install west==0.14.0
```

**Reply by TomzBench** [reply]
Still fails with same error. West still advertizes `0.14.0` so I hope i installed the hotfix correctly.

Here are logs again, including installation of venv and west so you can verify i test correctly.

https://gist.github.com/TomzBench/fc14d145075e3173baffcb246900ce68

**Comment by 424778940z** [comment]
I got the same issue, and solved it myself.
I'm running Windows 10 with Python 3.10.5, installed latest west which is 0.14

I did look into source, found this function
https://github.com/zephyrproject-rtos/west/blob/main/src/west/manifest.py#L861
```python
    def sha(self, rev: str, cwd: Optional[PathType] = None) -> str:
        '''Get the SHA for a project revision.
        :param rev: git revision (HEAD, v2.0.0, etc.) as a string
        :param cwd: directory to run command in (default:
            self.abspath)
        '''
        # Though we capture stderr, it will be available as the stderr
        # attribute in the CalledProcessError raised by git() in
        # Python 3.5 and above if this call fails.
        cp = self.git(f'rev-parse {rev}^{{commit}}', capture_stdout=True,
                      cwd=cwd, capture_stderr=True)
        # Assumption: SHAs are hex values and thus safe to decode in ASCII.
        # It'll be fun when we find out that was wrong and how...
        return cp.stdout.decode('ascii').strip()
```

If I get the idea right, it should give a result like `xxxxx^{commit}` that tells git the refer type is `commit` and hash is `xxxxx`?
My simple testing code gives me correct result, that means this is not a Python issue
```python
>>> rev:str = "53d3415c14d60f8f4bfca54bfbc5d5a667d7e724"
>>> print(f'rev-parse {rev}^{{commit}}')
rev-parse 53d3415c14d60f8f4bfca54bfbc5d5a667d7e724^{commit}
```
Then I checked with Procmon tool from SysinternalsSuite, the command that west invoake git is correct
I was using powershell with git that from Msys2 bin folder, seems git from Msys2 do not like it, and dropped `{}` around `commit` somehow
The solution is simple, either install a windows version git, or get a portable git and make a script put the folder containing it as the FIRST path in PATH

**Reply by TomzBench** [reply]
I confirm your solution fix for me as well.

I had windows git installed, but a cygwin git was favored on my path. so by changing my path variables order to favor the windows git, the solution worked as you said.
